### PR TITLE
Allow for more advanced substituting

### PR DESCRIPTION
**Pull request by seblyng**
Also removing an unnecessary `to_string()` from the last PR about cookie store

### CODE REVIEW

File: src/substitute.rs [hunk 1]
```diff
@@ -65,18 +65,27 @@ fn substitute_line(line: &str, env: &Table) -> Result<String> {
...
 
     let key = parts.next().unwrap_or("").trim();
-    match env.get(key) {
-        Some(Value::String(v)) => Ok(v.to_string()),
-        Some(Value::Integer(v)) => Ok(v.to_string()),
-        Some(Value::Float(v)) => Ok(v.to_string()),
-        Some(Value::Boolean(v)) => Ok(v.to_string()),
+    let parsed_key = key.chars().filter(valid_character).collect::<String>();
+
+    let parse = |v: &str| key.replace(&parsed_key, v);
```
Comment (ropez) [inline src/substitute.rs:79]: I'm totally confused. You'll need to explain what's going on here.

Comment (ropez) [inline src/substitute.rs:79]: I'm totally confused. You'll need to explain what's going on here.

Comment (seblyng) [reply src/substitute.rs:79]: I am collecting all `valid` characters of a name, to get the "real" name. Let's say I have this:

```
{
    "name": {{ "foo" | null }}
}
```

I will discard `"`, and `parsed_key` will be `foo` and not `"foo"`. Then inside the `parse`-closure, I am replacing only `foo`. So if I input `bar` to inquire, or I have extracted something to the `foo`-variable, the result will end up being

```
{
    "name": "bar"
}
```

and not

```
{
    "name": bar
}
```

Comment (seblyng) [reply src/substitute.rs:79]: It mostly means that I don't have to input the quotes to inquire

**Pull request merged**